.github/workflows/docker.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Publish
+
+on:
+  push:
+    branches: [ feature/docker-and-package-upgrades ]
+
+jobs:
+  build-and-push-docker-image:
+    # Publish to GitHub Container Registry
+    # Master branch only
+    # if: ${{ github.ref == 'refs/heads/master' }}
+    # needs:
+    #  - tests
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    env: 
+      REGISTRY: ghcr.io
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for OpenActive Test Suite
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/openactive/test-suite
+          labels: |
+            org.opencontainers.image.title=OpenActive Test Suite
+            org.opencontainers.image.description=Test suite for OpenActive data publishing and Open Booking API implementations
+            org.opencontainers.image.vendor=OpenActive
+          tags: |
+            latest
+
+      - name: Build and push Docker image for OpenActive Test Suite
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
This needs to be expanded to test before push (https://docs.docker.com/build/ci/github-actions/test-before-push/), to ensure that everything works as expected.

The reference implementation service (https://github.com/openactive/OpenActive.Server.NET/blob/d6e2f57dd8b6a7835cf8bcd747eef7553a7128f4/.github/workflows/docker-test.yaml#L16-L32) can be added here to facilitate this.

The config can also be passed in (https://github.com/openactive/OpenActive.Server.NET/blob/d6e2f57dd8b6a7835cf8bcd747eef7553a7128f4/.github/workflows/docker-test.yaml#L53-L56)

Though note that we'd need to get [ghcr.io/openactive/reference.bookingsystem.identity](http://ghcr.io/openactive/reference.bookingsystem.identity) running to actually test Puppeteer